src/lib/swapper/swappers/ThorchainSwapper/utils/longTailHelpers.ts

```diff
@@ -0,0 +1,55 @@
+import { avalancheChainId, bscChainId, ethChainId, fromAssetId } from '@shapeshiftoss/caip'
+import { Token } from '@uniswap/sdk-core'
+import type { Asset } from 'lib/asset-service'
+
+import type { ThornodePoolResponse } from '../types'
+import { WAVAX_TOKEN, WBNB_TOKEN, WETH_TOKEN } from './constants'
+
+export const getWrappedToken = (nativeAsset: Asset): Token => {
+  switch (nativeAsset.chainId) {
+    case ethChainId:
+      return WETH_TOKEN
+    case bscChainId:
+      return WBNB_TOKEN
+    case avalancheChainId:
+      return WAVAX_TOKEN
+    default:
+      throw new Error(`getWrappedToken: Unsupported chainId ${nativeAsset.chainId}`)
+  }
+}
+
+export const getTokenFromAsset = (asset: Asset): Token => {
+  const { symbol, name, precision, assetId } = asset
+  const chainReference = Number(fromAssetId(assetId).chainReference)
+  const assetReference = fromAssetId(assetId).assetReference
```

Comment:
fromAssetId can be be slow so we can minimise the number of calls
```suggestion
    const { chainReference: chainReferenceStr, assetReference } = fromAssetId(assetId)
    const chainReference = Number(chainReferenceStr)
```

Reply:
I like it: https://github.com/shapeshift/web/pull/5638/commits/27a1c79332b85efd53dde708777cf2687aec5322